src/stores/LifecycleStore.ts

```diff
@@ -51,6 +59,11 @@ class LifecycleStore extends AsyncStore<IState> {
                 });
                 break;
             case "MatrixActions.sync": {
+                if (payload.state === SyncState.Syncing && payload.prevState !== SyncState.Syncing) {
+                    // We've reconnected to the server: update server version support
+                    checkServerVersions();
```

Comment:
If you're deliberately firing an `async` operation off in the background, please say so and explain why. And preferably, add a `catch` handler to do something with errors rather than have them logged later as "unhandled exceptions".

Reply:
My initial thought was that it was odd to pop up a toast with the title "your server is unsupported", and meanwhile let the app proceed on its merry way.

But maybe I'm thinking about the toast wrong: it's not meant to stop the user using the app; rather it's meant as a gentle nudge to tell them to upgrade their server. Which is ok, but maybe we could write that in a comment so that other people don't have the same confusion.

Reply:
Yeah, fair - I skipped it in this case because it was a sync function so awaiting wasn't an option anyway, but more doc is more good.

Possibly, or to get their to nudge their HS admin to upgrade following an Element upgrade, I guess, and in the meantime they're not totally locked out of their app. I'll note it down.